support/Pipelines/Bbh/InitialData.py

```diff
@@ -0,0 +1,143 @@
+
+from spectre.support.Schedule import schedule, scheduler_options
+
+logger = logging.getLogger(__name__)
+
+ID_INPUT_FILE_TEMPLATE = Path(__file__).parent / "InitialData.yaml"
+
+
+def id_parameters(
+    mass_ratio: float,
+    separation: float,
+    orbital_angular_velocity: float,
+    refinement_level: int,
+    polynomial_order: int,
+):
+    """Determine initial data parameters from options.
+
+    These parameters fill the 'ID_INPUT_FILE_TEMPLATE'.
+
+    Arguments:
+      mass_ratio: Defined as q = M_A / M_B >= 1.
+      separation: Coordinate separation D of the black holes.
+      orbital_angular_velocity: Omega_0.
+      refinement_level: h-refinement level.
+      polynomial_order: p-refinement level.
+    """
+
+    # Sanity checks
+    assert mass_ratio >= 1.0, "Mass ratio is defined to be >= 1.0."
+
+    # Determine initial data parameters from options
+    M_A = mass_ratio / (1.0 + mass_ratio)
+    M_B = 1.0 / (1.0 + mass_ratio)
+    x_A = separation / (1.0 + mass_ratio)
+    x_B = x_A - separation
+    return {
+        "MassRight": M_A,
+        "MassLeft": M_B,
+        "XRight": x_A,
+        "XLeft": x_B,
+        "ExcisionRadiusRight": 0.89 * 2.0 * M_A,
+        "ExcisionRadiusLeft": 0.89 * 2.0 * M_B,
+        "OrbitalAngularVelocity": orbital_angular_velocity,
+        # Resolution
+        "L": refinement_level,
+        "P": polynomial_order,
+    }
+
+
+def generate_id(
+    mass_ratio: float,
+    separation: float,
+    orbital_angular_velocity: float,
+    refinement_level: int,
+    polynomial_order: int,
+    id_input_file_template: Union[str, Path] = ID_INPUT_FILE_TEMPLATE,
+    **scheduler_kwargs,
+):
+    """Generate initial data for a BBH simulation.
+
+    Parameters for the initial data will be inserted into the
+    'id_input_file_template'. The remaining options are forwarded to the
+    'schedule' command. See 'schedule' docs for details.
+    """
+    # Determine initial data parameters from options
+    id_params = id_parameters(
+        mass_ratio=mass_ratio,
+        separation=separation,
+        orbital_angular_velocity=orbital_angular_velocity,
+        refinement_level=refinement_level,
+        polynomial_order=polynomial_order,
+    )
+    logger.debug(f"Initial data parameters: {pretty_repr(id_params)}")
+
+    # Schedule!
+    return schedule(id_input_file_template, **id_params, **scheduler_kwargs)
+
+
+@click.command(name="generate-id", help=generate_id.__doc__)
+@click.option(
+    "--mass-ratio",
+    "-q",
+    type=float,
+    help="Mass ratio of the binary, defined as q = M_A / M_B >= 1.",
+)
+@click.option(
+    "--separation",
+    "-D",
+    type=float,
+    help="Coordinate separation D of the black holes.",
+)
+@click.option(
+    "--orbital-angular-velocity",
+    "-w",
+    type=float,
+    help="Orbital angular velocity Omega_0.",
+)
+@click.option(
+    "--refinement-level",
+    "-L",
+    type=int,
+    help="h-refinement level.",
+    default=0,
```

Comment:
Not to be figured out for this PR, but do we want to allow defaults? We explicitly don't in our executable input files for a variety of reasons, and this sort of gets around that.

Reply:
My plan for these resolution options is to replace them with something like "lev" or "target truncation error" once we have AMR. I can remove the default here for now, no strong opinion. BTW a big reason for avoiding defaults in input files is that they're so hard to discover and can change your simulation in subtle ways. Here they're just a `--help` away.